spec/deploy_bouncer_spec.rb

```diff
@@ -71,5 +71,103 @@
         end
       end
     end
+
+    it "fails when APP_DOMAIN is not set" do
+      ClimateControl.modify FASTLY_SERVICE_ID: "123321abc", FASTLY_API_KEY: "fastly@example.com" do
+        expect { DeployBouncer.new.deploy! }
+        .to raise_error(SystemExit, /APP_DOMAIN environment variable is not set/)
+      end
+    end
+
+    it "fails when FASTLY_SERVICE_ID is not set" do
+      ClimateControl.modify APP_DOMAIN: "gov.uk", FASTLY_API_KEY: "fastly@example.com" do
+        expect { DeployBouncer.new.deploy! }
+        .to raise_error(SystemExit, /FASTLY_SERVICE_ID environment variable is not set/)
+      end
+    end
+
+    it "fails when FASTLY_API_KEY is not set" do
+      ClimateControl.modify APP_DOMAIN: "gov.uk", FASTLY_SERVICE_ID: "123321abc" do
+        expect { DeployBouncer.new.deploy! }
+        .to raise_error(RuntimeError, /FASTLY_API_KEY is not set in the environment/)
+      end
+    end
+
+    it "fails when no hosts found in Transition hosts API" do
+      @requests = []
+
+      # Fastly#get_service. Return a service with two VCL "versions" (https://docs.fastly.com/api/config#version)
+      @requests << stub_request(:get, "https://api.fastly.com/service/123321abc")
+        .to_return(body: File.read("spec/fixtures/fastly-get-service-response.json"))
+
+      # We clone the latest active VCL version, which returns the latest version
+      @requests << stub_request(:put, "https://api.fastly.com/service/123321abc/version/2/clone")
+        .to_return(body: File.read("spec/fixtures/fastly-put-clone.json"))
```

Comment:
Looks like an opportunity to DRY this up, we can then make this test and the next one begin with something like:

```ruby
service_id = "123321abc"
@requests = stub_fastly_service(service_id: service_id)
# ...etc
```